device_doctor/lib/src/process_helper.dart

```diff
@@ -67,3 +69,50 @@ Future<String> eval(String executable, List<String> arguments,
 
   return output.trimRight();
 }
+
+/// Splits [from] into lines and selects those that contain [pattern].
+Iterable<String> grep(Pattern pattern, {@required String from}) {
```

Comment:
Can you add tests for these util functions?

Reply:
Added some tests for functions needing logic checks.